core/dictgen/src/BaseSelectionRule.cxx

```diff
@@ -226,27 +226,24 @@ BaseSelectionRule::EMatchType BaseSelectionRule::Match(const clang::NamedDecl *d
          return kName;
       }
    } else if (fHasNameAttribute) {
-      if (name_value == name) {
-         const_cast<BaseSelectionRule*>(this)->SetMatchFound(true);
-         return kName;
-      } else if ( !fCXXRecordDecl || fCXXRecordDecl != (void*)-1) {
-         // Possibly take the most expensive path if the fCXXRecordDecl is not
-         // set or we already took the expensive path and found nothing (-1).
-         const clang::CXXRecordDecl *target
-            = fHasFromTypedefAttribute ? nullptr : ROOT::TMetaUtils::ScopeSearch(name_value.c_str(), *fInterp,
-                                                   true /*diagnose*/, 0);
-
-         if ( target ) {
-            const_cast<BaseSelectionRule*>(this)->fCXXRecordDecl = target;
-         } else {
-            // If the lookup failed, let's not try it again, so mark the value has invalid.
-            const_cast<BaseSelectionRule*>(this)->fCXXRecordDecl = (clang::CXXRecordDecl*)-1;
-         }
-         if ( target && D && target == D ) {
-            const_cast<BaseSelectionRule*>(this)->SetMatchFound(true);
-            return kName;
-         }
-      }
+     if ((name_value == name) && (!fCXXRecordDecl || fCXXRecordDecl != (void*)-1)) {
+       // Possibly take the most expensive path if the fCXXRecordDecl is not
+       // set or we already took the expensive path and found nothing (-1).
+       const clang::CXXRecordDecl *target
+         = fHasFromTypedefAttribute ? nullptr : ROOT::TMetaUtils::ScopeSearch(name_value.c_str(), *fInterp,
+                                                                              true /*diagnose*/, 0);
+
+       if ( target ) {
+         const_cast<BaseSelectionRule*>(this)->fCXXRecordDecl = target;
+       } else {
+         // If the lookup failed, let's not try it again, so mark the value has invalid.
+         const_cast<BaseSelectionRule*>(this)->fCXXRecordDecl = (clang::CXXRecordDecl*)-1;
+       }
+     }
+     if ((name_value == name) || (fCXXRecordDecl && D && fCXXRecordDecl == D)) {
```

Comment:
Is the change really 'NFC'?  Doesn't this double the number of string comparisons?

Reply:
It does but I am trying to disentangle the code so that I can avoid the LookupHelper.

Reply:
Pushing a new version...

Reply:
That is fine.  I strongly recommend that we update the commit log to be clear (at the very least drop the NFC and possibly add the goal you are working towards).  Thanks.